aes/decryptor.go

```diff
@@ -0,0 +1,122 @@
+package aes
+
+import (
+	cryptoaes "crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type EncryptedValue struct {
+	data     []byte
+	iv       []byte
+	tag      []byte
+	datatype string
+}
+
+func parse(value string) (*EncryptedValue, error) {
+	re := regexp.MustCompile(`^ENC\[AES256_GCM,data:(.+),iv:(.+),tag:(.+),type:(.+)\]`)
+	matches := re.FindStringSubmatch(value)
+	if matches == nil {
+		return nil, fmt.Errorf("Input string %s does not match sops' data format", value)
+	}
+	data, err := base64.StdEncoding.DecodeString(matches[1])
+	if err != nil {
+		return nil, errors.New("Error base64-decoding data")
```

Comment:
make sure you return the actual error too: `fmt.Errorf("Error base64-decoding data: %v", err)`